starter-code/main.js

```diff
@@ -20,10 +20,82 @@ mongoClient.connect(url, (error, db) => {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "3":
+            //Contar todas las compañías por nombre fundadas en el 2004
+            db.collection('companies').find({ "funding_rounds.funded_year": 2004 }, { name: 1, _id: 0 }).count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "4":
+            //Listar por nombre, todas las compañías fundadas en febrero de 2004
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: 2 }] }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "5":
+            //Listar por nombre, todas las compañías fundadas en verano de 2004 (de abril a junio) y ordenadas por fecha
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: { $gte: 4 } }, { founded_month: { $lte: 6 } }] }, { name: 1, _id: 0 }).sort({ founded_month: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "6":
+            //¿Qué compañias tienen oficinas en Barcelona?
+            db.collection('companies').find({ offices: { $elemMatch: { city: "Barcelona" } } }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "7":
+            //Enumere las 10 empresas con más empleados ordenados de forma ascendente (mostrar nombre y empleados).
+              db.collection('companies').find({ number_of_employees: { $type: 'int' } }, { name: 1, _id: 0}).sort({number_of_employees: -1}).toArray((error, result) => {
```

Comment:
```suggestion
              db.collection('companies').find({ number_of_employees: { $type: 'int' } }, { name: 1, _id: 0}).limit(10).sort({number_of_employees: -1}).toArray((error, result) => {
```
Recuerda que con la opción Limit, puedes especificar el número máximo de documentos que se devolverán en una consulta.